modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -1136,18 +1135,37 @@ private static String generateParameterId(Parameter parameter) {
     }
 
     private Map<String, SecurityScheme> getAuthMethods(List<SecurityRequirement> securities, Map<String, SecurityScheme> securitySchemes) {
-        if (securities == null || (securitySchemes == null || securitySchemes.isEmpty())) {
+        if (securitySchemes == null || securitySchemes.isEmpty()) {
             return null;
         }
+
         final Map<String, SecurityScheme> authMethods = new HashMap<>();
-        for (SecurityRequirement requirement : securities) {
-            for (String key : requirement.keySet()) {
-                SecurityScheme securityScheme = securitySchemes.get(key);
-                if (securityScheme != null) {
-                    authMethods.put(key, securityScheme);
+
+        Components components       = openAPI.getComponents();
+        Map<String, Header> headers = (null != components) ? components.getHeaders() : null;
+
+        for (String key : securitySchemes.keySet()) {
+            SecurityScheme securityScheme = securitySchemes.get(key);
```

Comment:
@Yogarine What IDE did you use? Seems like it's aligning based on the "="

Reply:
IntelliJ IDEA. This is a personal preference though. I find it more readable when consecutive assignments are aligned.